reana_server/config.py

```diff
@@ -135,10 +134,32 @@ def _(x):
 APP_DEFAULT_SECURE_HEADERS["content_security_policy"] = {}
 APP_HEALTH_BLUEPRINT_ENABLED = False
 
-# Rate limiting configuration
+# Rate limiting configuration using invenio-app
 # ===========================
-RATELIMIT_AUTHENTICATED_USER = "20 per second"
-RATELIMIT_GUEST_USER = "20 per second"
+
+
+def _is_valid_rate_limit(rate_limit: str) -> bool:
+    return bool(
+        re.match(r"[0-9]+(\sper\s|\/)(second|minute|hour|day|month|year)", rate_limit,)
+    )
+
+
+def _get_rate_limit(env: str, default: str) -> str:
+    env_variable = os.getenv(env, default)
+    if _is_valid_rate_limit(env_variable):
+        logging.warning(
+            "Rate limit config string has incorrect format. "
+            f"Got: {env_variable}. Using default config: {default}"
+        )
+        return env_variable
```

Comment:
I'm not very sure this logic is correct. From reading it, I see that the warning is emitted if `_is_valid_rate_limit` is `True`?

In addition, how do you know that what was assigned to `env_variable` is the value coming from the env var or the default value?

Reply:
Still working on that. Need to change to draft.

Reply:
Done.